src/main/scala/ru/otus/sc/auth/data/dao/impl/AuthDaoImpl.scala

```diff
@@ -0,0 +1,61 @@
+package ru.otus.sc.auth.data.dao.impl
+
+import akka.http.scaladsl.util.FastFuture
+import ru.otus.sc.auth.data.dao.{AuthDao, TokenDao}
+import ru.otus.sc.auth.exceptions.{CredentialsNotUniqueException, LoginFailedException, TokenNotFoundException}
+import ru.otus.sc.auth.model.TokenView
+import ru.otus.sc.common.ImplicitHelpers
+import ru.otus.sc.user.data.{User, UserRepository}
+
+import scala.concurrent.Future
+
+class AuthDaoImpl extends AuthDao with ImplicitHelpers {
+
+  private lazy val userRepo           = UserRepository
+  private lazy val tokenDao: TokenDao = new TokenDaoImpl
+
+  override def auth(login: String, password: String): Future[TokenView] =
+    userRepo
+      .findAllByLogin(login)
+      .flatMap {
+        case user :: Nil => tokenDao.generateToken(user.login + user.password)
+        case Nil         => FastFuture.failed(LoginFailedException(login))
+        case user :: rest :: Nil =>
+          FastFuture.failed(CredentialsNotUniqueException("was detected not unique username"))
+      }
+      .map { token =>
+        TokenView(token.value, token.expiredAt)
+      }
+
+  override def register(
+      login: String,
+      password: String,
+      name: String,
+      email: String
+  ): Future[TokenView] =
+    tokenDao
```

Comment:
Писать нескльколько вложенных мапов-флетмапов не самая хорошая идея и вот почему - вложенность создает ограничения на легкость внесения изменений, потому как добавить ещё один вложенности куда-нибуть в середину задача не самая простая. К счастью, специально для Future есть синтаксис который называется for comprehension, который ковертирует невложенный код во вложенный: https://docs.scala-lang.org/tour/for-comprehensions.html 
```suggestion
    for {
      token <- tokenDao.generateToken(login + password)
      user <- userRepo.save(...)
    } yield(token, user)
```